src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -547,8 +545,10 @@ UInt SpatialPooler::initMapColumn_(UInt column) const {
 }
 
 
-vector<UInt> SpatialPooler::initMapPotential_(UInt column, bool wrapAround) {
-  NTA_ASSERT(column < numColumns_);
+void SpatialPooler::initMapPotential_(UInt column, bool wrapAround, SDR& potential) {
+  NTA_CHECK(potential.size == numInputs_);
+  NTA_CHECK(column < numColumns_);
```

Comment:
nitpick: assert.  This is a private method and is called from one place.

Reply:
ok, might prefer to keep it, it's just called during init, so once in a lifetime.